lib_xua/src/core/endpoint0/xua_endpoint0.c

```diff
@@ -1089,7 +1104,15 @@ void XUA_Endpoint0_loop(XUD_Result_t result, USB_SetupPacket_t sp, chanend c_ep0
                                         msos_desc_len = sizeof(MSOS_desc_composite_t);
                                     }
                                     memcpy(&desc_bos.usb_desc_bos_platform.CapabilityData[4], &msos_desc_len, sizeof(int16_t)); // Update msos descriptor length in platform capabilityData
-                                    result = XUD_DoGetRequest(ep0_out, ep0_in, (unsigned char*)&desc_bos, sizeof(USB_Descriptor_BOS_t), sp.wLength);
+                                    // On the Mac, BOS desc is requested twice, first with wLength 5 (just usb_desc_bos_standard), then with length 33 (usb_desc_bos_standard + ),
+                                    // while on Windows there's only one request of length 0xff.
+                                    // Set length appropriately
+                                    unsigned length = sp.wLength;
```

Comment:
Im surprised this is required, XUD_DoGetRequest() should handle this?

Reply:
I must have over corrected then. It wasn't obvious to me what XUD_DoGetRequest() would do when requested is greater than length. I'd like it to return length.

Reply:
it should send which ever is smaller

https://github.com/xmos/lib_xud/blob/65dc55a20789de08c9cd7503a80a9150fd006094/lib_xud/src/user/client/XUD_EpFunctions.xc#L33